Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -440,24 +441,25 @@ void LeptonTriggerSFProducer::Produce(ZJetEvent const& event,
                                  ZJetProduct& product,
                                  ZJetSettings const& settings) const
 {
-    if(product.m_zValid){
-        product.m_weights["zl1TriggerSFWeight"] = 1/GetScaleFactor(0, *product.m_zLeptons.first);
-        product.m_weights["zl2TriggerSFWeight"] = 1/GetScaleFactor(0, *product.m_zLeptons.second);
-        product.m_weights["leptonTriggerSFWeight"] =
-            1/(1-(1-GetScaleFactor(0, *product.m_zLeptons.first)) * (1-GetScaleFactor(0, *product.m_zLeptons.second)));
+    if(product.m_zValid) {
+        float sf1 = 1 / GetScaleFactor(0, *product.m_zLeptons.first);
+        float sf2 = 1 / GetScaleFactor(0, *product.m_zLeptons.second);
+        product.m_weights["zl1TriggerSFWeight"] = sf1;
+        product.m_weights["zl2TriggerSFWeight"] = sf2;
+        product.m_weights["leptonTriggerSFWeight"] = sf1 * sf2;
```

Comment:
This `1/(x*y)` is not the same as above: `1/(x*y-x-y)!

Reply:
I know, but the combination for the other SFs is simple mutliplication. Maybe the SFs for the Trigger were different at one point? And since it's just efficiencies we can multiply them.

Reply:
This needs to be verified before merging.

Reply:
Reverting this. Original is correct.
Since I use the SingleMuon trigger, the efficiency needs to be calculated differently:
Probability of the SingleMuon trigger not triggering is: (1-eff1)*(1-eff2), hence the SF is 1/(1-(1-eff1)(1-eff2))